src/navigator/types.tsx

```diff
@@ -40,6 +39,23 @@ interface SendConfirmationParams {
   feeTokenId?: string
 }
 
+interface SendEnterAmountParams {
+  recipient: Recipient & { address: string }
+  isFromScan: boolean
+  origin: SendOrigin
+  forceTokenId?: boolean
+  defaultTokenIdOverride?: string
+}
+
+interface ValidateRecipientParams {
+  transactionData?: TransactionDataInput
```

Comment:
Most of the logic hinges on this, now optional, param. If `transactionData` exists then we are in the old flow and should proceed to SendConfirmation, otherwise proceed to SendEnterAmount